lib/rspec/matchers/built_in/change.rb

```diff
@@ -87,8 +87,8 @@ def initialize(receiver=nil, message=nil, &block)
         end
 
         def raise_block_syntax_error
-          raise SyntaxError, "The block passed to the `change` matcher must " \
-            "use `{ ... }` instead of do/end"
+          raise SyntaxError, "Block passed to the wrong method. Perhaps you " \
```

Comment:
How about `Block not received by the `change` matcher.` Rather than to `the wrong method`, the later is ambigious as to where the message is coming from.